### PR TITLE
Add test to check vulnerable packages with triaged null

### PR DESCRIPTION
**Pull request by fedepacher**
|Related issue|
|-------------|
|     https://github.com/wazuh/wazuh-qa/issues/3420        |

## Description

This PR aims to add a test to check if vulnerability detection reads packages added with no triaged value after a partial scan.

### Added

- test_scan_vulnerabilities_triaged_null.py
- configuration_scan_vulnerabilities_triaged_null.yaml
- cases_scan_vulnerabilities_triaged_null.yaml

## Testing performed

| Tester             | Test path | Jenkins | Local  | OS | Commit | Notes                |
|--------------------|-----------|---------|--------|-----|--------|----------------------|
| @user (Developer)  |           | ⚫⚫⚫ | [:green_circle: ](https://github.com/wazuh/wazuh-qa/files/10007936/r1-3420.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/10007937/r2-3420.zip) [:green_circle:](https://github.com/wazuh/wazuh-qa/files/10007938/r3-3420.zip)  |         |         | Nothing to highlight |
| @user (Reviewer)   |           | ⚫⚫⚫ | :no_entry_sign: :no_entry_sign: :no_entry_sign:  |        |         | Nothing to highlight |

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+@pytest.mark.tier(level=1)
+@pytest.mark.parametrize('configuration, metadata, agent_system',
+                         zip(t1_configurations, t1_configuration_metadata, t1_systems), ids=t1_case_ids)
+def test_scan_triaged_null_vulnerabilities(configuration, metadata, agent_system, set_wazuh_configuration_vdt,
+                                           truncate_monitored_files, clean_cve_tables_func, prepare_scan,
+                                           restart_modulesd_function):
+    '''
+    description: Check if a vulnerable package with triaged NULL, is detected with the partial scan.
+
+    test_phases:
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py:124]: Use the new convention documentation:
- setup
- test
- tierdown

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+        - Set a custom Wazuh configuration.
+        - Mock an agent.
+        - Force a full scan.
+        - Restart wazuh-modulesd.
+        - Insert a vulberable package with triaged NULL.
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py:129]:
```suggestion
        - Insert a vulnerable package with triaged NULL.
```

File: tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_vulnerabilities_triaged_null.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+- name: RHEL
+  description: Check that after partial scan is launched triaged status changes from NULL to 1
+  configuration_parameters:
+    OS: '8'
+    SCAN_INTERVAL: '5'
+  metadata:
+    system: RHEL8
+    triaged: ''
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_scan_results/data/test_cases/cases_scan_vulnerabilities_triaged_null.yaml:8]: Is this  correct? `'' == None`?

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+
+    # Update packages sync status.
+    agent_db.update_sync_info(agent_id=agent_id, component="syscollector-packages")
+
+    # Check for the next partial scan
+    evm.check_partial_scan_start_log(agent_id=agent_id, timeout=vd.T_15)
+
+    query_string = f"agent {agent_id} sql SELECT triaged FROM sys_programs"
+
+    query_value = query_wdb(query_string)[0]
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py:189]: Encapsulate this in a framework method

**Review by jmv74211 — CHANGES_REQUESTED** [review]

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+    assert query_value['triaged'] == 1, f'The triaged value of sys_programs table should be 1 after a partial scan of '\
```
[the remark this answers is not in the document]
Comment (fedepacher) [reply tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py]: The query returns an `int` value

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py (hunk 4 above)
Comment (fedepacher) [reply tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3587/commits/840de2cd5fb93a74e75cb6bbcb6511b3ec15491b)

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py (hunk 2 above)
Comment (fedepacher) [reply tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py:139]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3587/commits/840de2cd5fb93a74e75cb6bbcb6511b3ec15491b)

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py (hunk 1 above)
Comment (fedepacher) [reply tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py:124]: Done in [here](https://github.com/wazuh/wazuh-qa/pull/3587/commits/840de2cd5fb93a74e75cb6bbcb6511b3ec15491b)

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+    - Mock an agent with a specified system.
+    - Force full scan.
+
+    Args:
+        metadata (dict): Test case metadata.
+        agent_system (str): System to set to the mocked agent.
+        mock_agent_with_custom_system (fixture): Mock an agent with a custom system.
+    """
+    # Forcing a full-scan
+    agent_db.update_last_full_scan(1, agent_id=mock_agent_with_custom_system)
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py:110]: Why are you forcing the full scan before launching the test?

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+                                                                                         f'partial scan of a '\
+                                                                                         f'vulnerable package.'
+
+    # Update packages sync status.
+    agent_db.update_sync_info(agent_id=agent_id, component="syscollector-packages")
+
+    # Check for the next partial scan
+    evm.check_partial_scan_start_log(agent_id=agent_id, timeout=vd.T_15)
+
+    assert agent_db.get_vulnerability_triaged(agent_id=agent_id) == 1, f'The triaged value of sys_programs table '\
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py:199]: Why are you checking for `int` number? You are using `str` in the other changes

File: tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+t1_systems = [metadata['system'] for metadata in t1_configuration_metadata]
+
+# Set offline custom feeds configuration
+t1_configurations = update_configuration_template(
+    t1_configurations,  ['CUSTOM_REDHAT_OVAL_FEED', 'CUSTOM_REDHAT_JSON_FEED', 'CUSTOM_NVD_JSON_FEED'],
+    [custom_redhat_oval_feed_path, custom_redhat_json_feed_path, custom_nvd_json_feed_path])
+
+
+@pytest.fixture(scope='function')
+def prepare_scan(metadata, agent_system, mock_agent_with_custom_system):
```
Comment (jmv74211) [inline tests/integration/test_vulnerability_detector/test_scan_results/test_scan_vulnerabilities_triaged_null.py:98]: You have to use a global fixture.

**Review by jmv74211 — CHANGES_REQUESTED** [review]

**Review by jmv74211 — APPROVED** [review]
